examples/plot_vector_testing.py

```diff
@@ -0,0 +1,158 @@
+    {"A": np.arange(5), "B": np.arange(5)}, geometry=gpd.points_from_xy(points.lon, points.lat), crs="epsg:4326"
+)
+point_gdf["attr2"] = ["Tree", "Tree", "Bush", "Bush", "Bush"]
+
+# Create symbology dictionary to use in the legend
+
+line_symb = {"road": "black", "stream": "blue"}
+point_symb = {"Tree": "green", "Bush": "brown"}
+
+################################################################################
+# Create Your Spatial Plot
+# -----------------------------------------------------------------
+# Above you created several GeoPandas GeoDataFrame objects that you want
+# to plot. To plot these data according to attribute value, you can group
+# the geometry by attributes and plot within a loop. Once you've created
+# your plot, you are ready to test it using MatPlotCheck.
+
+# Plot your data
+fig, ax = plt.subplots()
+polygon_gdf.plot(ax=ax, color="purple")
+
+# Plot your data by attributes using groupby
+for ctype, lines in multi_line_gdf.groupby('attr'):
+    color = line_symb[ctype]
+    label = ctype
+    lines.plot(color=color, ax=ax, label=label)
+
+size = 0
+
+for ctype, points in point_gdf.groupby('attr2'):
+    color = point_symb[ctype]
+    label = ctype
+    size += 100
+    points.plot(color=color, ax=ax, label=label, markersize=size)
+
+# Add a legend
+ax.legend(title="Legend", loc=(1.1, .1));
+
+# If you were running this in a notebook, the commented out  line below would
+# store the matplotlib object. However, in this example, you can just grab the
+# axes object directly.
+
+# plot_1_hold = nb.convert_axes(plt, which_axes="current")
+
+################################################################################
+# Create A VectorTester Object
+# --------------------------------
+# Once you've created your plot, you can create a MatPlotCheck VectorTester object
+# that can be used to test elements in the plot.
+
+vector_test = VectorTester(ax)
+
+###############################################################################
+#
+# .. note::
+#   Each geometry type must be tested seperately in VectorTester. So, if your
+#   plot has multiple geometry types, such as lines, polygons, and points,
+#   make sure to check each geometry type seperately.
+
+###############################################################################
+#
+# .. note::
+#   If a test fails, matplotcheck will return an error. If the test passes,
+#   no message is returned.
+
+################################################################################
+# Testing Point Attribute Values and Geometry
+# -------------------------------------------
+# You can check that both the position of the points on the plot and the associated
+# point attribute values are
+# accurate using assert_points(), assert_points_grouped_by_type() and
+# assert_collection_sorted_by_markersize. If the plot uses point markers that are
+# sized by attribute value, you can check that the
+# size of each marker correctly relates to an attribute value.
```

Comment:
maybe add why we are providing point_gdf to this method. Something like below you check that the plot contains the correct point locations by calling assert_points and providing the expected points data which in this case is the point_gdf object.